### DISCUSSION TITLE
CodeEditor component not rendering

### DISCUSSION THREAD

**Discussion by shrestha-bikash** [cloudscape-design/components#2681] [Q&A] [answered]
Hi team,

I am trying to use `CodeEditor` component following this guide https://cloudscape.design/components/code-editor?tabId=api but facing multiple issues. I have followed the exact example as shown in the guide. The first issue that I am seeing is: `Uncaught (in promise) Error: Unknown config key: useStrictCSP` and if I comment out this statement:`ace.config.set('useStrictCSP', true);`, the error goes away. But another error appears saying: `TypeError: ace.Range is not a constructor`. I am not able to find the root cause of this issue. Can someone help with this?

I am using `ace-build: 1.36.0`

**Comment by jperals** [comment, marked as answer]
Hi Bikash,

Sorry for the delay in the answer. Are you still facing this issue? If yes, can you first try upgrading Ace to the latest version (currently 1.39.0), and if it keeps happening, could you share a reproducible example or a snippet of your setup?
